docker/models/containers.py

```diff
@@ -709,7 +716,14 @@ def run(self, image, command=None, stdout=True, stderr=False,
         if exit_status != 0:
             stdout = False
             stderr = True
-        out = container.logs(stdout=stdout, stderr=stderr)
+
+        logging_driver = container.attrs['HostConfig']['LogConfig']['Type']
+
+        if logging_driver == 'json-file' or logging_driver == 'journald':
```

Comment:
I consider `client.containers.run` as replacement for `docker run` command. Docker run command does not fail if you run it with non readable driver. 

If you need different logic based on the error you are always able to run two commands:

```python
container = client.containers.run("busybox", detach=True, log_config=dict(type='none'))
container.logs()
Traceback (most recent call last):
.....
docker.errors.APIError: 500 Server Error: Internal Server Error ("configured logging driver does not support reading")
```

This pull request does not change `logs()` method behaviour, it affect only `run` method. Currently it is impossible to run attached container with non-readable logging driver.